src/TLuaInterpreter.cpp

```diff
@@ -10739,6 +10750,505 @@ int TLuaInterpreter::restartIrc(lua_State* L)
+    
+    return 0;
+}
+
+
+/** ttsSetSpeechRate( rate )
+ *  Sets the rate of speech playback.
+ */
+int TLuaInterpreter::ttsSetSpeechRate(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+    
+    float fRate;
+    if (!lua_isnumber(L, 1)) {
+        lua_pushfstring(L, "ttsSetSpeechRate: argument expected number, got %s!", luaL_typename(L, 1));
+        lua_error(L);
+        return 1;
+    } else {
+        fRate = lua_tonumber(L, 1);
+    }
+    
+    if (fRate > 1.0) {
+        fRate = 1.0;
+	}
+    
+	if (fRate < -1.0) {
+        fRate = -1.0;
+    }
+	
+    speechUnit->setRate(fRate);
+    
+    Host& host = getHostFromLua(L);
+    TEvent event;
+    event.mArgumentList.append(QLatin1String("ttsRateChanged"));
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+    event.mArgumentList.append(QString::number(fRate));
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    host.raiseEvent(event);
+    
+    return 0;
+}
+
+
+/** ttsSetSpeechPitch( pitch )
+ *  Sets the pitch of speech playback.
+ */
+int TLuaInterpreter::ttsSetSpeechPitch(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+    
+    float fPitch;
+    if (!lua_isnumber(L, 1)) {
+        lua_pushfstring(L, "ttsSetSpeechPitch: argument expected number, got %s!", luaL_typename(L, 1));
+        lua_error(L);
+        return 1;
+    } else {
+        fPitch = lua_tonumber(L, 1);
+    }
+    
+    if (fPitch > 1.0) {
+        fPitch = 1.0;
+	}
+    
+    if (fPitch < -1.0) {
+        fPitch = -1.0;
+	}
+    
+    speechUnit->setPitch(fPitch);
+    
+    TEvent event;
+    Host& host = getHostFromLua(L);
+    event.mArgumentList.append(QLatin1String("ttsPitchChanged"));
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+    event.mArgumentList.append(QString::number(fPitch));
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    host.raiseEvent(event);
+    
+    return 0;
+}
+
+
+/** ttsSetSpeechVolume( volume )
+ *  Sets the volume of speech playback.
+ */
+int TLuaInterpreter::ttsSetSpeechVolume(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+    
+    float fVol;
+    if (!lua_isnumber(L, 1)) {
+        lua_pushfstring(L, "ttsSetSpeechVolume: argument expected number, got %s!", luaL_typename(L, 1));
+        lua_error(L);
+        return 1;
+    } else {
+        fVol = lua_tonumber(L, 1);
+    }
+    
+	if (fVol > 1.0) {
+        fVol = 1.0;
+    }
+	
+    if (fVol < 0.0) {
+        fVol = 0.0;
+    }
+	
+    speechUnit->setVolume(fVol);
+    
+    Host& host = getHostFromLua(L);
+    TEvent event;
+    event.mArgumentList.append(QLatin1String("ttsVolumeChanged"));
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+    event.mArgumentList.append(QString::number(fVol));
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    host.raiseEvent(event);
+    
+    return 0;
+}
+
+
+/** ttsGetVoices()
+ *  Lists all voices available to the current OS locale.
+ *  Currently uses the default system locale.
```

Comment:
You might like to note that #1334 will (when I get it finished) record the user's choice of locale/language for GUI purposes although it will, unless manually changed, follow their settings for their "Desktop" by default which __can be a range of values__ (which the Qt libraries will not tell us which one was actually used when a GUI translation is selected from that range - see [QTBUG-36608](https://bugreports.qt.io/browse/QTBUG-36608) which is duplicated by my [QTBUG-64416](https://bugreports.qt.io/browse/QTBUG-64416)).